test/hooks.on-ready.test.js

```diff
@@ -0,0 +1,277 @@
+'use strict'
+
+const t = require('tap')
+const Fastify = require('../fastify')
+const wait = require('util').promisify(setTimeout)
+
+t.test('onReady should be called in order', t => {
+  t.plan(7)
+  const fastify = Fastify()
+
+  let order = 0
+
+  fastify.addHook('onReady', function (done) {
+    t.equals(order++, 0, 'called in root')
+    t.equals(this.pluginName, fastify.pluginName, 'the this binding is the right instance')
+    done()
+  })
+
+  fastify.register(async (childOne, o) => {
+    childOne.addHook('onReady', function (done) {
+      t.equals(order++, 1, 'called in childOne')
+      t.equals(this.pluginName, childOne.pluginName, 'the this binding is the right instance')
+      done()
+    })
+
+    childOne.register(async (childTwo, o) => {
+      childTwo.addHook('onReady', async function () {
+        await wait(100)
+        t.equals(order++, 2, 'called in childTwo')
+        t.equals(this.pluginName, childTwo.pluginName, 'the this binding is the right instance')
+      })
+    })
+  })
+
+  fastify.ready(err => t.error(err))
+})
+
+t.test('async onReady should be called in order', async t => {
+  t.plan(7)
+  const fastify = Fastify()
+
+  let order = 0
+
+  fastify.addHook('onReady', async function () {
+    await wait(100)
```

Comment:
I would prefer if we could not wait for 100ms each time. This makes this test last for seconds.

```js
const immediate = promisify(setImmediate)
```

Has the same effect on the event loop.

Reply:
Thanks to pointing it out 👍